lib/portage/dbapi/bintree.py

```diff
@@ -143,23 +146,40 @@ def aux_get(self, mycpv, wants, myrepo=None):
 		if not self.bintree._remotepkgs or \
 			not self.bintree.isremote(mycpv):
 			try:
-				tbz2_path = self.bintree._pkg_paths[instance_key]
+				binpkg_path = self.bintree._pkg_paths[instance_key]
 			except KeyError:
 				raise KeyError(mycpv)
-			tbz2_path = os.path.join(self.bintree.pkgdir, tbz2_path)
+			binpkg_path = os.path.join(self.bintree.pkgdir, binpkg_path)
 			try:
-				st = os.lstat(tbz2_path)
+				st = os.lstat(binpkg_path)
 			except OSError:
 				raise KeyError(mycpv)
-			metadata_bytes = portage.xpak.tbz2(tbz2_path).get_data()
+			binpkg_format = mycpv._metadata.get('BINPKG_FORMAT', None)
+			if not binpkg_format:
+				binpkg_format = get_binpkg_format(binpkg_path)
```

Comment:
At the beginning of `aux_get`, `self.bintree.populate()` populates `self.cpvdict[instance_key]['BINPKG_FORMAT']`, so we can simplify the logic by assuming that `self.cpvdict[instance_key]['BINPKG_FORMAT']` always has the correct `BINPKG_FORMAT` here.